### PR TITLE
refactor: various ante handler improvements

### PR DESCRIPTION
**Pull request by johnletey**
This PR improves the FiatTokenFactory [Ante Handler](https://docs.cosmos.network/main/learn/advanced/baseapp#antehandler) used in Noble today.

For those not aware, ante handlers are used to accept/reject transactions on a Cosmos SDK chain.

The FiatTokenFactory uses them to check if $USDC is paused or if the sender/receiver of $USDC is blacklisted.

## Breakdown of Changes

In aa5e5c15159cf68cb154c4df90fd45cea6d28809, we fix the ante handlers by adding support for [Authz](https://docs.cosmos.network/main/build/modules/authz) messages.

In 06375f8c4cb4a55d9b797081949fd2d76f59369b, we improve the ante handler by correctly checking if we can decode the IBC receiver address. As the $USDC blacklist only applies to Noble, when sending to other chains via IBC, we can skip the check if we can't decode the address. Thanks to @yito88 and the [Heliax](https://heliax.dev) team for reporting this!

In a4baa478a812b6497d2b6a8d017304d03adf769e, we improve the ante handlers by making them reusable in other codebases (like the Noble chain) — making this repository the true truth of source.

Additionally, developer tooling was updated to the latest version (with zero effect on the codebase) to improvement development experience!

### CODE REVIEW

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ on:
   push:
     branches: [master]
 
+permissions:
+  contents: write
```
Comment (yvonnezhangc) [inline .github/workflows/unit-tests.yaml:9]: the addition of this line seems to be causing test failures. @johnletey Could you take a look?

**Comment by yvonnezhangc** [comment]
@johnletey thanks for raising this PR. From relevant docs I understand why it may be beneficial to enable support for authz messages, but could you also provide some more context regarding why this change is urgent and impactful for Noble? How widely has this support been adopted in the cosmos ecosystem? Are there relevant examples we can look at for further reference?

File: x/fiattokenfactory/ante.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			for _, o := range m.Outputs {
+				for _, c := range o.Coins {
+					err := checkForBlacklistedAddressByTokenFactory(ctx, o.Address, c, ad.fiattokenfactory)
+					if errors.Is(err, fiattokenfactorytypes.ErrUnauthorized) {
+						return sdkerrors.Wrapf(err, "an address (%s) is blacklisted and can not send or receive tokens", o.Address)
+					} else if err != nil {
+						return sdkerrors.Wrapf(err, "error decoding address (%s)", o.Address)
+					}
+				}
+			}
```
Comment (yvonnezhangc) [inline x/fiattokenfactory/ante.go:157]: The code blocks that handle inputs & outputs seem to be highly repetitive. Any way to consolidate?

**Comment by yvonnezhangc** [comment]
I don't see new / modified tests as a part of this PR - how are we testing the changes?

File: x/fiattokenfactory/ante.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+			if errors.Is(err, fiattokenfactorytypes.ErrUnauthorized) {
+				return sdkerrors.Wrapf(err, "an address (%s) is blacklisted and can not send tokens", m.Sender)
+			} else if err != nil {
+				return sdkerrors.Wrapf(err, "error decoding address (%s)", m.Sender)
+			}
+			err = checkForBlacklistedAddressByTokenFactory(ctx, m.Receiver, m.Token, ad.fiattokenfactory)
+			if errors.Is(err, fiattokenfactorytypes.ErrUnauthorized) {
+				return sdkerrors.Wrapf(err, "an address (%s) is blacklisted and can not receive tokens", m.Receiver)
+			} else if err != nil {
+				// Ignore the decoding error because the receiver address could be a non-cosmos address and the destination chain should be in charge of minting the token
```
Comment (yvonnezhangc) [inline x/fiattokenfactory/ante.go:169]: Curious, do all destination chains have privilege to mint tokens? Is there a way to enforce a blacklist check on a destination chain that uses non-cosmos addresses?

File: x/fiattokenfactory/ante.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
+package fiattokenfactory
```
Comment (yvonnezhangc) [inline x/fiattokenfactory/ante.go:1]: No objection to code restructuring, but curious how these functions would be used outside fiattokenfactory. Any existing use cases you have in mind?

**Comment by johnletey** [comment]
These changes were implemented in #22 (14edf83ee1c96055e2c17ea56ca9dd303d3c14f6).

**Pull request closed without merging**